File: leo2-central/src/test/java/org/deku/leo2/central/data/NodeJooqRepositoryTest.java
package org.deku.leo2.central.data;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.deku.leo2.central.DataTest;
import org.deku.leo2.central.data.entities.jooq.tables.records.MstNodeRecord;
import org.deku.leo2.central.data.repositories.jooq.NodeJooqRepository;
import org.junit.Test;

import javax.inject.Inject;

/**
 * Created by masc on 02.07.15.
 */
public class NodeJooqRepositoryTest extends DataTest {
    private Log mLog = LogFactory.getLog(this.getClass());

    @Inject
    NodeJooqRepository mNodeJooqRepository;

    @Test
    public void testNodeJooqRepository() {
        MstNodeRecord r = mNodeJooqRepository.saveByKey("test2", "555", "456");

        mLog.info(String.format("Node id [%d]", r.getNodeId()));
    }
}

File: leoz-central/src/test/java/org/deku/leoz/central/data/NodeJooqRepositoryTest.kt
package org.deku.leoz.central.data

import org.deku.leoz.central.DataTest
import org.deku.leoz.central.data.jooq.tables.records.MstNodeRecord
import org.deku.leoz.central.data.repository.NodeJooqRepository
import org.junit.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

import javax.inject.Inject

/**
 * Created by masc on 02.07.15.
 */
class NodeJooqRepositoryTest : DataTest() {
    private val log = LoggerFactory.getLogger(this.javaClass)

    @Inject
    private lateinit var nodeJooqRepository: NodeJooqRepository

    @Test
    fun testNodeJooqRepository() {
        val r = nodeJooqRepository.findByKeyOrCreateNew("test2")

        log.info(String.format("Node id [%d]", r.nodeId))
    }
}
